packages/decoder/lib/Web3Adapter.ts

```diff
@@ -0,0 +1,50 @@
+import Web3Eth from "web3-eth";
+type PastLogsOptions = {
+  toBlock?: string | number;
+  fromBlock?: string | number;
+  address?: string | string[];
+}
+import type BN from "bn.js";
+
+export class Web3Adapter {
+  public provider: any;
```

Comment:
Something I just realized - earlier today in Slack, @eggplantzzz, you said that direct use of `provider` requires that we separately handle websockets. Why is that? Wouldn't `.send()`, et al. work the same, regardless of http vs. ws? @davidmurdoch I guess you were also part of the discussion? I feel like I missed an important detail there!

Reply:
I was referring to the case of implementing your own internal provider (like using `axios` to send rewuests) you'd need to also handle the websocket case.